src/index.ts

```diff
@@ -317,6 +320,23 @@ export function createClient (args: InitArgs): Client {
     })
   }
 
+  const lookupTransfers = async (batch: TransferID[]): Promise<Transfer[]> => {
+    return new Promise((resolve, reject) => {
+      const callback = (error: undefined | Error, results: Transfer[]) => {
+        if (error) {
+          reject(error)
+        }
+        resolve(results)
```

Comment:
@DonChangfoot is it correct that we have always done a fall-through here for the error path, when rejecting? Should we not do `return reject(error)` or I think I'm probably misunderstanding i.e. we must both reject and resolve when there's an error? Just want to double-check. Thanks!

Reply:
@jorangreef good catch! The `resolve` won't change the state of the promise once `reject` is called. But we would want to `return` after the `reject` to prevent further code execution.

Reply:
Awesome. Thanks guys. I updated all of the callbacks to have `return` after `reject`.

Reply:
Marking as resolved for now.

Reply:
Thanks @DonChangfoot !